### PR TITLE
Upgrade clickhouse image in tests

### PR DESCRIPTION
**Pull request by sylph-eu**
## Description
I'm going to invest some time into making ClickHouse connector more functional. This is the first PR in a (hopefully) longer series.

Changes:
- Upgrade ClickHouse docker images (both by ClickHouse and Altinity). I was picking lts/stable builds from both providers.
- Fix unit tests to let them work against newer version. I had to disable one case in `testDouble` due to [bug in ClickHouse itself](https://github.com/ClickHouse/ClickHouse/issues/60146).
- Changed the way IP addresses are casted upon ingestion from Trino. Previous implementation couldn't handle NULL values properly.

## Additional context and related issues
None.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java [hunk 1]
```diff
@@ -29,14 +29,14 @@
 public class TestingClickHouseServer
         implements Closeable
 {
-    private static final DockerImageName CLICKHOUSE_IMAGE = DockerImageName.parse("yandex/clickhouse-server");
-    public static final DockerImageName CLICKHOUSE_LATEST_IMAGE = CLICKHOUSE_IMAGE.withTag("21.11.10.1");
-    public static final DockerImageName CLICKHOUSE_DEFAULT_IMAGE = CLICKHOUSE_IMAGE.withTag("21.8.14.5"); // EOL is 31 Aug 2022
+    private static final DockerImageName CLICKHOUSE_IMAGE = DockerImageName.parse("clickhouse/clickhouse-server");
+    public static final DockerImageName CLICKHOUSE_LATEST_IMAGE = CLICKHOUSE_IMAGE.withTag("24.1.8.22");
+    public static final DockerImageName CLICKHOUSE_DEFAULT_IMAGE = CLICKHOUSE_IMAGE.withTag("23.8.12.13");
```
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java:34]: Could you leave a code comment of EOL?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java [hunk 2]
```diff
@@ -29,14 +29,14 @@
...
 
     // Altinity Stable Builds Life-Cycle Table https://docs.altinity.com/altinitystablebuilds/#altinity-stable-builds-life-cycle-table
-    private static final DockerImageName ALTINITY_IMAGE = DockerImageName.parse("altinity/clickhouse-server").asCompatibleSubstituteFor("yandex/clickhouse-server");
-    public static final DockerImageName ALTINITY_LATEST_IMAGE = ALTINITY_IMAGE.withTag("21.8.13.1.altinitystable");
-    public static final DockerImageName ALTINITY_DEFAULT_IMAGE = ALTINITY_IMAGE.withTag("20.8.4.11_aes"); // EOL is 02 December 2022
+    private static final DockerImageName ALTINITY_IMAGE = DockerImageName.parse("altinity/clickhouse-server").asCompatibleSubstituteFor("clickhouse/clickhouse-server");
+    public static final DockerImageName ALTINITY_LATEST_IMAGE = ALTINITY_IMAGE.withTag("23.8.8.21.altinitystable");
+    public static final DockerImageName ALTINITY_DEFAULT_IMAGE = ALTINITY_IMAGE.withTag("22.8.20.12.altinitystable"); // EOL is 13 Feb 2026
```
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java:39]: Is it possible to change to 21.8? 
Screenshot 2024-04-10 at 8 05 43

**Review by ebyhr — APPROVED** [review]
Please squash commits into one and follow the commit message guideline: https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

**Review by ebyhr — COMMENTED** [review]
Please update `Requirements` in `clickhouse.md`.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java (hunk 1 above)
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java:34]: Done.

Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java:34]: Did you forget pushing the change?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java (hunk 2 above)
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java:39]: Done, although 21.8 docker image doesn't yet support Mac/Arm "out of the box", left comment about it.
Newer versions do work both on Intel and Arm: https://altinity.com/blog/announcing-arm64-altinity-stable-builds-for-clickhouse

**Comment by sylph-eu** [comment]
I've signed CLA and sent it out yesterday.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Please update the commit title to "Update testing ClickHouse image version" and remove commit body.

**Comment by sylph-eu** [comment]
Re-pushed commit to re-trigger build due to my inclusion into cla list.

**Review by wendigo — APPROVED** [review]

**Pull request merged**